airflow/providers/snowflake/transfers/snowflake_to_slack.py

```diff
@@ -14,72 +14,144 @@
 
     :param sql: The SQL statement to execute on Snowflake (templated)
-    :param slack_message: The templated Slack message to send with the data returned from Snowflake.
-        You can use the default JINJA variable {{ results_df }} to access the pandas dataframe containing the
-        SQL results
+    :param use_slack_webhook: Use legacy Slack WebHook connection.
+    :param filename: If specified than result send to slack as file, rather than as text.
+        Only support SlackHook connection. Filename should contain valid extension which referenced
+        to format: csv, parquet, json, etc. It is also possible to set
+        compression in extension: csv.gzip, parquet.snappy, etc. If filename contains not supported
+        file formats than operator fall back to uncompressed csv. (templated)
+    :param slack_message: Use the templated Slack message to send with the data returned from Snowflake if
+        ``filename`` not specified. You can use the default Jinja variable {{ results_df }} to access the
+        pandas dataframe containing the SQL results. (templated)
     :param snowflake_conn_id: Reference to
         :ref:`Snowflake connection id<howto/connection:snowflake>`
-    :param slack_conn_id: The connection id for Slack
-    :param results_df_name: The name of the JINJA template's dataframe variable, default is 'results_df'
+    :param slack_conn_id: The connection id for Slack. Either Slack API or Slack Incoming Webhooks
+        could use, control by `use_slack_webhook`.
     :param parameters: The parameters to pass to the SQL query
     :param warehouse: The Snowflake virtual warehouse to use to run the SQL query
     :param database: The Snowflake database to use for the SQL query
     :param schema: The schema to run the SQL against in Snowflake
     :param role: The role to use when connecting to Snowflake
-    :param slack_token: The token to use to authenticate to Slack. If this is not provided, the
-        'webhook_token' attribute needs to be specified in the 'Extra' JSON field against the slack_conn_id
+    :param results_df_name: The name of the JINJA template's dataframe variable, default is 'results_df'
+    :param df_to_params: The parameters to pass to ``pandas.Dataframe.to_<file_format>`` method.
+        See: https://pandas.pydata.org/docs/reference/io.html
+    :param channels: Comma-separated list of channel names or IDs where the file will be shared.
+        If not specified than file would upload to Slack Workspace files only.
+    :param initial_comment: The message text introducing the file in specified ``channels``.
+    :param slack_token: The token to use to authenticate to Slack. If this is not provided,
+        than use token from connection.
     """
 
-    template_fields: Sequence[str] = ('sql', 'slack_message')
-    template_ext: Sequence[str] = ('.sql', '.jinja', '.j2')
+    SUPPORTED_FILE_FORMATS = (
+        'csv',
+        'parquet',
```

Comment:
I dont think sending parquet to slack is a reasonable use case?

Reply:
I you ask personally me, than answer very clear "No, I don't think sending parquet to slack is a reasonable"

But users want to do some strange things and ability to send parquet cost on developing side almost nothing. This is the reason why I keep it.